elide-core/src/main/java/com/yahoo/elide/Elide.java

```diff
@@ -85,12 +93,60 @@ public Elide(AuditLogger auditLogger, DataStore dataStore) {
      * @param dictionary the dictionary
      * @param mapper Serializer/Deserializer for JSON API
      */
+    @Deprecated
     public Elide(AuditLogger auditLogger, DataStore dataStore, EntityDictionary dictionary, JsonApiMapper mapper) {
+        this(auditLogger, dataStore, dictionary, mapper, ActivePermissionExecutor.class);
+    }
+
+    /**
+     * Instantiates a new Elide.
+     *
+     * @param auditLogger the audit logger
+     * @param dataStore the dataStore
+     * @param dictionary the dictionary
+     * @param mapper Serializer/Deserializer for JSON API
+     * @param permissionExecutor Custom permission executor implementation
+     */
+    private Elide(AuditLogger auditLogger,
+                  DataStore dataStore,
+                  EntityDictionary dictionary,
+                  JsonApiMapper mapper,
+                  Class<? extends PermissionExecutor> permissionExecutor) {
         this.auditLogger = auditLogger;
         this.dataStore = dataStore;
         this.dictionary = dictionary;
         dataStore.populateEntityDictionary(dictionary);
         this.mapper = mapper;
+        this.permissionExecutor = permissionExecutor;
+    }
+
+    /**
+     * Elide Builder for constructing an Elide instance.
+     */
+    public static class Builder {
+        private final AuditLogger auditLogger;
+        private final DataStore dataStore;
+        @Getter @Setter private EntityDictionary entityDictionary;
+        @Getter @Setter private JsonApiMapper jsonApiMapper;
+        @Getter @Setter private Class<? extends PermissionExecutor> permissionExecutor;
+
+        /**
+         * Constructor.
+         *
+         * @param auditLogger
+         * @param dataStore
+         */
+        public Builder(AuditLogger auditLogger, DataStore dataStore) {
```

Comment:
Generally the builder pattern looks like

Builder().dictionary(new Dictionary()).jsonApiMapper(new Mapper()).permissionExecuter(() -> new MyPermissionExecuter()).build();